modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassSystemMenu.java

```diff
@@ -169,32 +214,14 @@ private void insertMenu(final Menu parent, final MenuBase mb, int pos) {
                     }
                 }
             }
-        });
-
-        for (MenuItemBase item : items) {
-            if (item instanceof MenuBase) {
-                // submenu
-                addMenu(glassMenu, (MenuBase)item);
-            } else {
-                // menu item
-                addMenuItem(glassMenu, item);
-            }
-        }
-        glassMenu.setPixels(getPixels(mb));
-
-        setMenuBindings(glassMenu, mb);
-
-        if (parent != null) {
-            parent.insert(glassMenu, pos);
-        } else {
-            glassSystemMenuBar.insert(glassMenu, pos);
-        }
+        };
     }
 
-    private void setMenuBindings(final Menu glassMenu, final MenuBase mb) {
-        mb.textProperty().addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
-        mb.disableProperty().addListener(valueModel -> glassMenu.setEnabled(!mb.isDisable()));
-        mb.mnemonicParsingProperty().addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
+
+    protected void setMenuBindings(final Menu glassMenu, final MenuBase mb) {
+        mb.textProperty().when(active).addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
+        mb.disableProperty().when(active).addListener(valueModel -> glassMenu.setEnabled(!mb.isDisable()));
+        mb.mnemonicParsingProperty().when(active).addListener(valueModel -> glassMenu.setTitle(parseText(mb)));
```

Comment:
is `active` guaranteed to be non-null at this point (and also on L279)?

assuming the code on L98 is fixed and does not clobber the old pointer.